plutus-use-cases/src/Plutus/Contracts/Auction.hs

```diff
@@ -264,25 +263,43 @@ data BuyerEvent =
 waitForChange :: SlotConfig -> AuctionParams -> StateMachineClient AuctionState AuctionInput -> HighestBid -> Contract AuctionOutput BuyerSchema AuctionError BuyerEvent
 waitForChange slotCfg AuctionParams{apEndTime} client lastHighestBid = do
     t <- currentTime
+
+    let targetTime = TimeSlot.slotToBeginPOSIXTime slotCfg
```

Comment:
can we use `apEndTime` as the target time? That way we don't need to loop as much.

(If this is a bigger change we could do it as a separate PR)

Reply:
Doing that results in an infinite loop. Recommend creating an issue for a separate PR.